pydm/data_plugins/plugin.py

```diff
@@ -113,6 +115,13 @@ def receive_from_channel(self, payload):
         pass
 
     def send_to_channel(self):
+        time = datetime.datetime.now()
+        if self._last_notify is None:
+            diff_time = 1
+        else:
+            diff_time = (time - self._last_notify).total_seconds()
+        self._notify_freq = round(1.0 / max(1e-6, diff_time), 0)
```

Comment:
I think the frequency might be more confusing than it is worth. If a widget is updating at a high rate,  but then suddenly disconnects the reported frequency will remain high. I understand that you can check the last notify time to see this is not the case, but I think a lot of people will have difficulty parsing this. I either suggest putting the frequency on a timer outside of `send_to_channel` such that we can see it go to zero upon disconnection or not reporting it at all.

Reply:
I think it could be more useful than it is harmful, but I can definitely see your point about confusion.

How about instead just stashing up to X previous timestamps and dynamically updating the notify_freq (as in a `property`)?